### PR TITLE
fix(*): KubeArmor goes brrr... ⚡⚡

### PR DESCRIPTION
**Pull request by daemon1024**
This PR handles of lots of edge cases which incur a lot of Performance load on KubeArmor.
Please go through individual commit messages to understand what has changed.
This in conjuction with #1491 and #1505 have brought down kubearmor CPU usage from >1000Mi to <100Mi ⚡

### CODE REVIEW

File: KubeArmor/BPF/system_monitor.c [hunk 1]
```diff
@@ -408,81 +430,55 @@ static __always_inline void get_outer_key(struct outer_key *pokey,
...
-#else // MONITOR_CONTAINER or MONITOR_CONTAINER_AND_HOST
+// == Pid NS Management == //
 
+static __always_inline u32 add_pid_ns()
+{
+    struct task_struct *task = (struct task_struct *)bpf_get_current_task();
+    u32 one = 1;
     u32 pid_ns = get_task_pid_ns_id(task);
-    if (pid_ns == PROC_PID_INIT_INO)
+    if (get_kubearmor_config(_MONITOR_HOST) && pid_ns == PROC_PID_INIT_INO)
```
Comment (nyrahul) [inline KubeArmor/BPF/system_monitor.c]: I think we should reverse the checks for performance reasons.
```suggestion
    if (pid_ns == PROC_PID_INIT_INO && get_kubearmor_config(_MONITOR_HOST))
```

File: KubeArmor/BPF/system_monitor.c [hunk 2]
```diff
@@ -408,81 +430,55 @@ static __always_inline void get_outer_key(struct outer_key *pokey,
...
     { // host
         u32 pid = bpf_get_current_pid_tgid() >> 32;
-        if (bpf_map_lookup_elem(&pid_ns_map, &pid) != 0)
+        if (!bpf_map_lookup_elem(&pid_ns_map, &pid))
```
Comment (nyrahul) [inline KubeArmor/BPF/system_monitor.c:454]: Isn't this change completely changing the semantics? reversing the check.

Comment (nyrahul) [reply KubeArmor/BPF/system_monitor.c:454]: Oh I see the code below and it makes sense

File: KubeArmor/BPF/system_monitor.c [hunk 3]
```diff
@@ -324,6 +324,28 @@ struct visibility
...
 
+#define DEFAULT_VISIBILITY_KEY 0xc0ffee
+
+// == Config == //
+
+enum
+{
+    _MONITOR_HOST = 0,
+    _MONITOR_CONTAINER = 1,
+    _ENFORCER_BPFLSM = 2,
```
Comment (nyrahul) [inline KubeArmor/BPF/system_monitor.c:335]: Having MONITOR and ENFORCER in the same enum might be a problem in the future. Won't we need enforcer and Monitor enabled at the same time?

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: KubeArmor/BPF/system_monitor.c (hunk 3 above)
Comment (daemon1024) [reply KubeArmor/BPF/system_monitor.c:335]: These enums are not values but indexes of the map.

All three of the values are configured independently and serve different usecases. It's not an if else case.

**Comment by daemon1024** [comment]
Closes https://github.com/kubearmor/KubeArmor/issues/1356

**Review by nyrahul — APPROVED** [review]
LGTM

**Pull request merged**